### PR TITLE
Fixed #28257 -- Confirmed support for GDAL 2.2.

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

**Comment by timgraham** [comment]
Maybe some comments could elaborate on which GDAL versions have which characteristics so this could be simplified as support for those versions is dropped?

**Comment by claudep** [comment]
I can add a note about the AUTHORITY addition in GDAL 2.2, but I don't really know about the 1984 vs 84 change.

**Comment by timgraham** [comment]
If GDAL is the determining factory in 84/1984, then it might be a change in GDAL 1.9 based on the timetable of https://code.djangoproject.com/ticket/18795 (reported with GDAL 1.9 and 8 months after its release). Django is supporting GDAL 1.9+ now, so maybe we can try dropping support for "84".

File: tests/gis_tests/gdal_tests/test_ds.py [hunk 1]
```diff
@@ -9,17 +10,26 @@
...
 
+wgs_84_wkt = (
+    'GEOGCS["GCS_WGS_1984",DATUM["WGS_1984",SPHEROID["WGS_1984",'
+    '6378137,298.257223563]],PRIMEM["Greenwich",0],UNIT["Degree",'
+    '0.017453292519943295]]'
+)
+# Using a regex because of small differences depending on GDAL versions.
+# AUTHORITY part has been added in GDAL 2.2.
+wgs_84_wkt_regex = (
+    r'^GEOGCS\["GCS_WGS_1984",DATUM\["WGS_1984",SPHEROID\["WGS_1984",'
```
Comment (timgraham) [inline tests/gis_tests/gdal_tests/test_ds.py]: It seems that the third "WGS_1984" on this line is normalized to "WGS_84". With that change, this is passing for me on ubuntu 14.04/16.04.

**Comment by timgraham** [comment]
Is there a reason not to also include the GDAL 2.2 docs changes in this commit?

**Pull request closed without merging**